File: app/src/main/java/com/example/firstapplication/ActivityThree.java
package com.example.firstapplication;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.firstapplication.DB.DBHelper;

public class ActivityThree extends AppCompatActivity {

    private TextView mTextView;
    public static int timeStampPiked;
    public static Day chosenDay;
    DBHelper dbHelper = new DBHelper(this);
    boolean updateData = false;

    EditText etSubject, etType, etPlace, etTeacher, etDay, etStart, etEnd;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_three);

        mTextView = (TextView) findViewById(R.id.text);

        etSubject = (EditText) findViewById(R.id.inputSubject2);
        etType = (EditText) findViewById(R.id.inputType);
        etPlace = (EditText) findViewById(R.id.inputPlace);
        etTeacher = (EditText) findViewById(R.id.inputTeacher);
        etDay = (EditText) findViewById(R.id.inputDay);
        etStart = (EditText) findViewById(R.id.inputStartTime);
        etEnd = (EditText) findViewById(R.id.inputEndTime);

        loadTimeStamp();
        checkInDB();
    }

    public void loadTimeStamp(){
        SQLiteDatabase database = dbHelper.getWritableDatabase();

        TimeFrame chosenTimeframe = TimeFrame.BASE_TIMEFRAMES[timeStampPiked];
        mTextView = (TextView) findViewById(R.id.inputStartTime);
        mTextView.setText(chosenTimeframe.get_start().get_timeInString());
        mTextView = (TextView) findViewById(R.id.inputEndTime);
        mTextView.setText(chosenTimeframe.get_end().get_timeInString());

        mTextView = (TextView) findViewById(R.id.inputDay);
        mTextView.setText(chosenDay.toString());
    }

    public void checkInDB(){
        SQLiteDatabase database = dbHelper.getWritableDatabase();

        Cursor cursor = database.query(DBHelper.NAME_TABLE_SCHEDULE,null,null,null,null, null,null);
        if (cursor.moveToFirst()) {
            int oneIndex = cursor.getColumnIndex(DBHelper.KEY_SUBJECT);
            int twoIndex = cursor.getColumnIndex(DBHelper.KEY_TYPE);
            int threeIndex = cursor.getColumnIndex(DBHelper.KEY_PLACE);
            int fourIndex = cursor.getColumnIndex(DBHelper.KEY_TEACHER);

            int forCheckOne = cursor.getColumnIndex(DBHelper.KEY_DAY);
            int forCheckTwo = cursor.getColumnIndex(DBHelper.KEY_LESSON_TIME);

            mTextView = (TextView) findViewById(R.id.inputDay);

            do {
                Toast.makeText(this, (cursor.getString(forCheckOne) + " to "+ chosenDay.toString()) + " - " + (cursor.getInt(forCheckTwo) + " to " + timeStampPiked), Toast.LENGTH_SHORT).show();

                if ((cursor.getString(forCheckOne).equals(chosenDay.toString())) && (cursor.getInt(forCheckTwo) == timeStampPiked))
                {
                    Toast.makeText(this, "Works", Toast.LENGTH_SHORT).show();
                    Log.d("mLog", " -" + cursor.getString(forCheckOne) + "- and -" + mTextView.getText() + "-");
                    mTextView = (TextView) findViewById(R.id.inputSubject2);
                    mTextView.setText(cursor.getString(oneIndex));
                    mTextView = (TextView) findViewById(R.id.inputType);
                    mTextView.setText(cursor.getString(twoIndex));
                    mTextView = (TextView) findViewById(R.id.inputPlace);
                    mTextView.setText(cursor.getString(threeIndex));
                    mTextView = (TextView) findViewById(R.id.inputTeacher);
                    mTextView.setText(cursor.getString(fourIndex));
                    updateData = true;

                    findViewById(R.id.button3).setVisibility(View.VISIBLE);
                    break;
                }

            } while (cursor.moveToNext());
        }
        cursor.close();
        dbHelper.close();
    }

    public int DaySwapper(String s1)
    {
        switch (s1)
        {
            case "Понедельник": return 1;
            case "Вторник": return 2;
            case "Среда": return 3;
            case "Четверг": return 4;
            case "Пятница": return 5;
            case "Суббота": return 6;
            default:
                Log.d("mLog", "ОШИБОЧКА");
                return -1;
        }
    }

    @SuppressLint("NonConstantResourceId")
    public void onClick(View v) {
        Intent intent;
        switch (v.getId())   {
            case R.id.buttonBack:
                intent = new Intent(this,ActivityTwo.class);
                startActivity(intent);
                break;
            case R.id.buttonSave:
                if ((etSubject.getText().length() > 0) &&
                        (etType.getText().length() > 0) &&
                        (etPlace.getText().length() > 0) &&
                        (etTeacher.getText().length() > 0) &&
                        (etDay.getText().length() > 0) &&
                        (etStart.getText().length() > 0) &&
                        (etEnd.getText().length() > 0)) {
                    SQLiteDatabase database = dbHelper.getWritableDatabase();
                    ContentValues contentValues = new ContentValues();
                    ContentValues contentValues1 = new ContentValues();

                    // сначала записать данные в отдельные таблицы, затем в одну большую
                    String subject = etSubject.getText().toString();
                    contentValues.put(DBHelper.KEY_VALUE, subject);
                    database.insert(DBHelper.TABLE_SUBJECTS, null, contentValues);
                    contentValues.clear();

                    String type = etType.getText().toString();
                    contentValues.put(DBHelper.KEY_VALUE, type);
                    database.insert(DBHelper.NAME_TABLE_TYPES, null, contentValues);
                    contentValues.clear();

                    String place = etPlace.getText().toString();
                    contentValues.put(DBHelper.KEY_VALUE, place);
                    database.insert(DBHelper.TABLE_PLACES, null, contentValues);
                    contentValues.clear();

                    String teacher = etTeacher.getText().toString();
                    contentValues.put(DBHelper.KEY_VALUE, teacher);
                    database.insert(DBHelper.TABLE_TEACHERS, null, contentValues);
                    contentValues.clear();

                    int day = DaySwapper(etDay.getText().toString());
                    String start = etStart.getText().toString();
                    String end = etEnd.getText().toString();

                    contentValues1.put(DBHelper.KEY_SUBJECT, subject);
                    contentValues1.put(DBHelper.KEY_TYPE, type);
                    contentValues1.put(DBHelper.KEY_PLACE, place);
                    contentValues1.put(DBHelper.KEY_TEACHER, teacher);
                    contentValues1.put(DBHelper.KEY_DAY, etDay.getText().toString());
                    contentValues1.put(DBHelper.KEY_LESSON_TIME, timeStampPiked);

                    if (updateData)  database.update(DBHelper.NAME_TABLE_SCHEDULE, contentValues1, DBHelper.KEY_LESSON_TIME + "= "+ timeStampPiked + " AND " + DBHelper.KEY_DAY +"= " + day, null);
                    else  database.insert(DBHelper.NAME_TABLE_SCHEDULE, null, contentValues1);

                dbHelper.close();
                intent = new Intent(this,ActivityTwo.class);
                startActivity(intent);
                }
                break;
            case R.id.delDay:
                mTextView = (TextView) findViewById(R.id.inputDay);
                mTextView.setText("");
                break;
            case R.id.delEndTime:
                mTextView = (TextView) findViewById(R.id.inputEndTime);
                mTextView.setText("");
                break;
            case R.id.delPlace:
                mTextView = (TextView) findViewById(R.id.inputPlace);
                mTextView.setText("");
                break;
            case R.id.delStartTime:
                mTextView = (TextView) findViewById(R.id.inputStartTime);
                mTextView.setText("");
                break;
            case R.id.delSubject2:
                mTextView = (TextView) findViewById(R.id.inputSubject2);
                mTextView.setText("");
                break;
            case R.id.delTeacher:
                mTextView = (TextView) findViewById(R.id.inputTeacher);
                mTextView.setText("");
                break;
            case R.id.delType:
                mTextView = (TextView) findViewById(R.id.inputType);
                mTextView.setText("");
                break;
            case R.id.button3:
                SQLiteDatabase database = dbHelper.getWritableDatabase();
                database.delete(DBHelper.NAME_TABLE_SCHEDULE, DBHelper.KEY_LESSON_TIME + "= "+ timeStampPiked + " AND " + DBHelper.KEY_DAY +"= " + chosenDay, null);
                dbHelper.close();
                intent = new Intent(this,ActivityFour.class);
                startActivity(intent);
                break;
            default:
                intent = new Intent(this,MainActivity.class);
                startActivity(intent);
                break;
        }
    }
}

File: app/src/main/java/com/example/firstapplication/ActivityThree.kt
package com.example.firstapplication

import android.annotation.SuppressLint
import android.content.ContentValues
import android.content.Intent
import android.database.sqlite.SQLiteDatabase
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import com.example.firstapplication.DB.DBHelper
import com.example.firstapplication.databinding.ActivityThreeBinding
import com.example.firstapplication.model.*

class ActivityThree : AppCompatActivity() {
    private lateinit var mTextView: TextView

    var dbHelper = DBHelper(this)
    var updateData = false

    private lateinit var binding: ActivityThreeBinding
    private lateinit var adapter: ColourAdapter

    lateinit var etSubject: EditText
    lateinit var etType: EditText
    lateinit var etPlace: EditText
    lateinit var etTeacher: EditText
    lateinit var etDay: EditText
    lateinit var etStart: EditText
    lateinit var etEnd: EditText
    lateinit var etNotification: EditText
    private val colourService: ColourService
        get() = (applicationContext as App).colourService

    private val cellsService: CellsService
        get() = (applicationContext as App).cellsService

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityThreeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        adapter = ColourAdapter(object :  ColourActionListener{
            override fun OnPick(colour: Colour) {
                // надо это подготовить для баазы данных, но пока этого нет
                pickedColour = android.graphics.Color.argb(255,colour.red, colour.green, colour.blue)
                findViewById<View>(R.id.colors).setBackgroundColor(pickedColour)
            }
        })

        try {
            //mTextView = findViewById<View>(R.id.text) as TextView
            etSubject = findViewById<View>(R.id.inputSubject2) as EditText
            etType = findViewById<View>(R.id.inputType) as EditText
            etPlace = findViewById<View>(R.id.inputPlace) as EditText
            etTeacher = findViewById<View>(R.id.inputTeacher) as EditText
            etDay = findViewById<View>(R.id.inputDay) as EditText
            etStart = findViewById<View>(R.id.inputStartTime) as EditText
            etEnd = findViewById<View>(R.id.inputEndTime) as EditText
            // пока нет ничего так как нет соответсвующего поля
        // etNotification = findViewById<View>(R.id.inputEndTime) as EditText
        } catch ( e : Exception) {
            Toast.makeText(this, "starting init doesn't work", Toast.LENGTH_SHORT).show()
        }
        try {
            loadTimeStamp()
            checkInDB()
        } catch (e : Exception){
            Toast.makeText(this, "advanced init doesn't work", Toast.LENGTH_SHORT).show()
        }

        val layoutManager = GridLayoutManager(this,6)
        binding.recyclerView.layoutManager = layoutManager
        binding.recyclerView.adapter = adapter


        colourService.addListener(colourListener)
    }

    private val colourListener : ColourListener = {
        adapter.colours = it
    }

    override fun onDestroy() {
        super.onDestroy()
        colourService.removeListener(colourListener)
    }

    fun loadTimeStamp() {
        val chosenTimeframe = TimeFrame.BASE_TIMEFRAMES[timeStampPiked]
        FillTextView(etStart, chosenTimeframe._start._timeInString)
        FillTextView(etEnd, chosenTimeframe._end._timeInString)
        FillTextView(etDay, chosenDay.toString())
    }

    fun checkInDB() {
        val database = dbHelper.writableDatabase
        val cursor = database.query(DBHelper.NAME_TABLE_SCHEDULE, null, null, null, null, null, null)
        if (cursor.moveToFirst()) {
            val oneIndex = cursor.getColumnIndex(DBHelper.KEY_SUBJECT)
            val twoIndex = cursor.getColumnIndex(DBHelper.KEY_TYPE)
            val threeIndex = cursor.getColumnIndex(DBHelper.KEY_PLACE)
            val fourIndex = cursor.getColumnIndex(DBHelper.KEY_TEACHER)
            val forCheckOne = cursor.getColumnIndex(DBHelper.KEY_DAY)
            val forCheckTwo = cursor.getColumnIndex(DBHelper.KEY_LESSON_TIME)
            mTextView = findViewById<View>(R.id.inputDay) as TextView
            do {
                if (cursor.getString(forCheckOne) == chosenDay.toString() && cursor.getInt(forCheckTwo) == timeStampPiked) {
                    FillTextView(etSubject, cursor.getString(oneIndex))
                    FillTextView(etType, cursor.getString(twoIndex))
                    FillTextView(etPlace, cursor.getString(threeIndex))
                    FillTextView(etTeacher, cursor.getString(fourIndex))
                    findViewById<View>(R.id.colors).setBackgroundColor(pickedColour)
                    updateData = true
                    findViewById<View>(R.id.del_input_button).visibility = View.VISIBLE
                    break
                }
            } while (cursor.moveToNext())
        }
        cursor.close()
        dbHelper.close()
    }

    fun FillTextView(textView: TextView, value: String) {
        textView.text = value
    }


    fun DaySwapper(s1: String?): Int {
        return when (s1) {
            "Понедельник" -> 1
            "Вторник" -> 2
            "Среда" -> 3
            "Четверг" -> 4
            "Пятница" -> 5
            "Суббота" -> 6
            else -> {
                Log.d("mLog", "ОШИБОЧКА")
                -1
            }
        }
    }

    fun putDataInDirectory(database : SQLiteDatabase ,tableName : String, text: EditText) {
        val textValue = text.text.toString()
        val contentValues = ContentValues()
        contentValues.put(DBHelper.KEY_VALUE, textValue)
        database.insert(DBHelper.TABLE_SUBJECTS, null, contentValues)
        contentValues.clear()
    }

    fun fillContentValues(contentValues: ContentValues, keys: Array<String>, contentArray: Array<EditText>) {
        for (i in 0..keys.count()) {
            contentValues.put(keys[i],contentArray[i].text.toString())
        }
    }

    @SuppressLint("NonConstantResourceId")
    fun onClick(v: View) {
        val intent: Intent
        when (v.id) {
            R.id.buttonBack -> {
                intent = Intent(this, ActivityTwo::class.java)
                startActivity(intent)
            }
            R.id.buttonSave -> if (etSubject.text.isNotEmpty() &&
                etType.text.isNotEmpty() &&
                etPlace.text.isNotEmpty() &&
                etTeacher.text.isNotEmpty()
            ) {
                val database = dbHelper.writableDatabase
                val contentValues = ContentValues()
                val editTexts = arrayOf<EditText>(etSubject, etType, etPlace, etTeacher, etDay)
                val directories = arrayOf<String>(DBHelper.TABLE_SUBJECTS, DBHelper.NAME_TABLE_TYPES, DBHelper.TABLE_PLACES, DBHelper.TABLE_TEACHERS)
                val keys = arrayOf<String>(DBHelper.KEY_SUBJECT, DBHelper.KEY_TYPE, DBHelper.KEY_PLACE, DBHelper.KEY_TEACHER, DBHelper.KEY_DAY)

                try {
                    // сначала записать данные в отдельные таблицы, затем в одну большую
                    for (i in 0..3) {
                        putDataInDirectory(database, directories[i], editTexts[i])
                    }
                    //COLUMN_ID, COLUMN_DAY, COLUMN_LESSON_TIME, COLUMN_SUBJECT, COLUMN_TYPE, COLUMN_TEACHER, COLUMN_PLACE, COLUMN_COLOR, COLUMN_NOTIFICATION

                    for (i in 0 until keys.count()) {
                        contentValues.put(keys[i], editTexts[i].text.toString())
                    }
                    //fillContentValues(contentValues,keys,editTexts)
                    contentValues.put(DBHelper.KEY_LESSON_TIME, timeStampPiked)
                    contentValues.put(DBHelper.KEY_NOTIFICATION, 0)
                    contentValues.put(DBHelper.KEY_COLOR, pickedColour)
                    try {
                        if (updateData) database.update(
                            DBHelper.NAME_TABLE_SCHEDULE,
                            contentValues,
                            DBHelper.KEY_LESSON_TIME + "= " + timeStampPiked + " AND " + DBHelper.KEY_DAY + "= " + etDay.text.toString(),
                            null
                        ) else database.insert(DBHelper.NAME_TABLE_SCHEDULE, null, contentValues)
                        dbHelper.close()
                        intent = Intent(this, ActivityTwo::class.java)
                        startActivity(intent)
                    } catch (e: Exception) {
                        Toast.makeText(this, "database insertion problem", Toast.LENGTH_SHORT).show()
                        Toast.makeText(this, "${e.message}", Toast.LENGTH_SHORT).show()
                    }
                } catch (e: Exception) {
                    Toast.makeText(this, "content value input problem", Toast.LENGTH_SHORT).show()
                    Toast.makeText(this, "${e.message}", Toast.LENGTH_SHORT).show()
                }
            }
            R.id.delDay, R.id.delEndTime, R.id.delPlace, R.id.delStartTime, R.id.delSubject2, R.id.delTeacher, R.id.delType -> {
                mTextView = findViewById<View>(v.id) as TextView
                mTextView.text = ""
            }
            R.id.del_input_button -> {
                try {
                    val database = dbHelper.writableDatabase
                    database.delete(
                        DBHelper.NAME_TABLE_SCHEDULE,
                        DBHelper.KEY_LESSON_TIME + "= " + timeStampPiked + " AND " + DBHelper.KEY_DAY + "= '" + chosenDay + "'",
                        null
                    )
                    dbHelper.close()
                    cellsService.clearCell(chosenDay, timeStampPiked)
                } catch (e: Exception) {
                    Toast.makeText(this, "deletion wasn't successful", Toast.LENGTH_SHORT).show()
                }
                intent = Intent(this, ActivityTwo::class.java)
                startActivity(intent)
            }
            else -> {
                intent = Intent(this, MainActivity::class.java)
                startActivity(intent)
            }
        }
    }

    companion object {
        var timeStampPiked = -1
        var chosenDay: Day = Day.Sunday
        var pickedColour = 0
    }
}